coderd/workspaceagents.go

```diff
@@ -70,7 +70,7 @@ func (api *API) workspaceAgentDial(rw http.ResponseWriter, r *http.Request) {
 
 	workspaceAgent := httpmw.WorkspaceAgentParam(r)
 	workspace := httpmw.WorkspaceParam(r)
-	if !api.Authorize(r, rbac.ActionUpdate, workspace) {
+	if !api.Authorize(r, rbac.ActionCreate, workspace.ExecutionRBAC()) {
```

Comment:
I didn't quite catch why this changed from Update to Create. My intuition says it doesn't really matter. So it's probably fine but just checking.

Reply:
It doesn't. It's arbitrary.

But what I did change was I added `ResourceWorkspaceExecution` as a unique resource. So giving CRUD to a workspace **does not** grant execution via a tunnel. These are now distinct permissions.